core/src/test/scala/pureconfig/ConfigCursorSuite.scala

```diff
@@ -39,24 +39,33 @@ class ConfigCursorSuite extends BaseSuite {
 
   it should "allow being casted to a list cursor in a safe way" in {
     cursor("abc").asListCursor should failWith(
-      WrongType(ConfigValueType.STRING, Set(ConfigValueType.LIST)), defaultPathStr)
+      WrongType(ConfigValueType.STRING, Set(ConfigValueType.LIST, ConfigValueType.OBJECT)), defaultPathStr)
 
     cursor("[1, 2]").asListCursor shouldBe
       Right(ConfigListCursor(conf("[1, 2]").asInstanceOf[ConfigList], defaultPath))
 
     cursor("{ a: 1, b: 2 }").asListCursor should failWith(
-      WrongType(ConfigValueType.OBJECT, Set(ConfigValueType.LIST)), defaultPathStr)
+      CannotConvertObjectToList(List("a", "b")), defaultPathStr)
+
+    cursor("{ 0: a, 1: b }").asListCursor shouldBe
+      Right(ConfigListCursor(conf("""["a", "b"]""").asInstanceOf[ConfigList], defaultPath))
+
+    cursor("{ 10: a, 3: b }").asListCursor shouldBe
+      Right(ConfigListCursor(conf("""["b", "a"]""").asInstanceOf[ConfigList], defaultPath))
   }
 
   it should "allow being casted to a list of cursors in a safe way" in {
     cursor("abc").asList should failWith(
-      WrongType(ConfigValueType.STRING, Set(ConfigValueType.LIST)), defaultPathStr)
+      WrongType(ConfigValueType.STRING, Set(ConfigValueType.LIST, ConfigValueType.OBJECT)), defaultPathStr)
```

Comment:
It seems strange to provide an error saying that we were expecting a `LIST` or an `OBJECT` when we used `asList`. I do not have a better idea, but it is a bit misleading for those who do not know these auto-conversions in HOCON (many people, I believe).

This is related to my comment above regarding typing. We are trying to make strong AST typing and auto-conversions coexist peacefully, which is bound to lead to strange decisions in the error reporting and handling aspect.

Reply:
I also had some doubts about this. I suppose it's OK to always say we were expecting a `LIST` when the conversion failed. It's more informative and less confusing for people who are not aware of the automatic conversions.